modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useStateSafe.ts

```diff
@@ -14,15 +14,15 @@
 
 import React from 'react';
 
-import useIsMounted from './useIsMounted.es';
+import useIsMounted from './useIsMounted';
 
 const {useCallback, useState} = React;
 
 /**
  * Wrapper for `useState` that does an `isMounted()` check behind the scenes
  * before triggering side-effects.
  */
-export default function useStateSafe(initialValue) {
+export default function useStateSafe<T = unknown>(initialValue: T | (() => T)) {
```

Comment:
Blargh! 😂

Reply:
It's actually quite nice. TS can infer the type:

```ts
const x = useStateSafe(1);
```

ie. it knows that `x` is `number`.

And you can also tell it if it doesn't know:

```ts
const x = useStateSafe<MyFancyType>(thing);
```

The `unknown` is good, because it forces you to deal with unsoundness problems if you pass in something without type information; eg.

```ts
function Component({randomCrap}: {[key: string]: unknown}) {
  const x = useStateSafe(randomCrap);
}
```

Here TS will complain if you try to use `x` anywhere where `unknown` isn't safe.

Reply:
I see! 👍